### PR TITLE
add carroussel

### PR DESCRIPTION
**Pull request by worms19**
To make the interface more user-friendly  putting all the image in a carrousel is easier for people to navigate.

This PR is open to improvement for later developpement because you can add several carroussel in a same page (classing by name, category etc ...)

### CODE REVIEW

File: frontend/src/components/VideoCarrousel.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        </div>
+    </Slide>
+});
+
+    return (    <CarouselProvider
+            visibleSlides={4}
+            totalSlides={videos.length}
+            step={2}
+            naturalSlideWidth={400}
+            naturalSlideHeight={500}
```
Comment (DerouineauNicolas) [inline frontend/src/components/VideoCarrousel.js]: This resizing of the thumbnails doesn't keep the original video aspect ratio. At the moment, the thumbnails are created with a [320,  320 x 16/9] size in the  [backend](https://github.com/Web-Multi-Media/HttpStreamingServer/blob/master/backend/StreamServerApp/utils.py#L134). So I think we should use the same height on the frontend.

File: frontend/src/api/djangoAPI.js [hunk 2]
```diff
@@ -2,6 +2,6 @@
 import axios from 'axios';
 
 export default axios.create({
-    //baseURL: 'http://localhost:1337/streaming/',
-    baseURL: '/streaming/',
+    baseURL: 'http://localhost:1337/streaming/',
+    //baseURL: '/streaming/',
```
Comment (xavierfav) [inline frontend/src/api/djangoAPI.js]: We should really find a convenient solution for automatically changing the API end point (e.g. [here](https://medium.com/@trekinbami/using-environment-variables-in-react-6b0a99d83cf5)?).
Using and environment variable for production and having a default one that is used in dev would be nice.

File: frontend/src/components/App.js [hunk 3]
```diff
@@ -53,14 +53,22 @@ class App extends React.Component {
                 <div className='ui grid'>
                     <div className="ui column">
                         <div className="eleven wide row">
-                            <VideoDetail video={this.state.selectedVideo} />
-                        </div>
-                        <div className="five wide row">
-                            <VideoList handleVideoSelect={this.handleVideoSelect} videos={this.state.videos} />
+                            <VideoDetail video={this.state.selectedVideo}  />
                         </div>
+
```
Comment (xavierfav) [inline frontend/src/components/App.js]: extract space here

File: frontend/src/components/App.js [hunk 4]
```diff
@@ -53,14 +53,22 @@ class App extends React.Component {
...
+                <div>
+                    {
+                        this.state.videos.length > 0 &&
+                        <div>
+                            <VideoCarrousel videos = {this.state.videos} handleVideoSelect={this.handleVideoSelect} />
+                        </div>
+                    }
+                </div>
             </div>
+
```
Comment (xavierfav) [inline frontend/src/components/App.js]: and here

File: frontend/src/components/VideoCarrousel.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                        {slider}
+                    </Slider>
+                }
+            </div>
+            <ButtonFirst>First</ButtonFirst>
+            <ButtonBack>Back</ButtonBack>
+            <ButtonNext>Next</ButtonNext>
+            <ButtonLast>Last</ButtonLast>
+        </CarouselProvider>
+
```
Comment (xavierfav) [inline frontend/src/components/VideoCarrousel.js]: fix indents in this part

File: frontend/src/components/VideoList.js [hunk 6]
```diff
@@ -10,6 +11,8 @@ const VideoList = ({videos , handleVideoSelect}) => {
         return <VideoItem key={video.pk} video={video} handleVideoSelect={handleVideoSelect} />
     });
 
+
```
Comment (xavierfav) [inline frontend/src/components/VideoList.js]: extra space here

File: frontend/src/style/style.scss [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+  top: 50%;
+  left: 50%;
+  height: 45%;
+  width: 100%;
+  text-align: center;
+  padding: 0px 5px 0px 5px;
+  overflow-wrap: break-word;
+  transform: translate(-50%, -50%);
+  -ms-transform: translate(-50%, -50%);
+}
```
Comment (xavierfav) [inline frontend/src/style/style.scss]: I understand this css file is taken from another code.
Using it as our main css file would make further development a bit more complicated.
Could we easily discard the things we don't need here?

Comment (worms19) [reply frontend/src/style/style.scss]: done

File: frontend/src/components/VideoCarrousel.js (hunk 1 above)
Comment (worms19) [reply frontend/src/components/VideoCarrousel.js]: done

File: frontend/src/components/App.js (hunk 3 above)
Comment (worms19) [reply frontend/src/components/App.js]: done

File: frontend/src/components/App.js (hunk 4 above)
Comment (worms19) [reply frontend/src/components/App.js]: done

File: frontend/src/components/VideoCarrousel.js (hunk 5 above)
Comment (worms19) [reply frontend/src/components/VideoCarrousel.js]: done

File: frontend/src/components/VideoList.js (hunk 6 above)
Comment (worms19) [reply frontend/src/components/VideoList.js]: done

File: frontend/src/style/style.scss (hunk 7 above)
Comment (worms19) [reply frontend/src/style/style.scss]: done

File: frontend/src/components/VideoCarrouselSlick.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import djangoAPI from "../api/djangoAPI";
+import SampleNextArrow from "./SampleNextArrow";
+import SamplePrevArrow from "./SamplePrevArrow";
+import '../style/style.scss';
+
+
+class VideoCarrouselSlick extends Component {
+
+    CARROUSSEL_SIZE = 5;
+    VIDEO_PER_PAGE = 10;
```
Comment (xavierfav) [inline frontend/src/components/VideoCarrouselSlick.js]: Would be great not to hardcode this parameter.
This is a configuration in the backend. If we change it, we have to change it in both places.
So we should find a way that the client gets this parameters from the backend.
Could we for instance get this parameter from the first search request send to the server?

File: frontend/src/components/VideoCarrouselSlick.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+            slidesToScroll: 5,
+            nextArrow: <SampleNextArrow />,
+            prevArrow: <SamplePrevArrow />,
+            afterChange: current => this.afterChangeMethod(current)
+        };
+
+        const slider = this.state.videos.map((video, vIndex) => {
+            return <div><img className='img-cover' onClick={() => this.props.handleVideoSelect(video)} src={video.fields.thumbnail} />
+                <p className='paragraph'>{video.fields.name}</p></div>
+
```
Comment (xavierfav) [inline frontend/src/components/VideoCarrouselSlick.js]: extra spaces here

File: frontend/src/components/VideoCarrouselSlick.js [hunk 10]
```diff
@@ -8,7 +8,6 @@ import '../style/style.scss';
 
 class VideoCarrouselSlick extends Component {
 
-    CARROUSSEL_SIZE = 5;
     VIDEO_PER_PAGE = 10;
```
Comment (xavierfav) [inline frontend/src/components/VideoCarrouselSlick.js]: We should avoid using this hardcoded param here. What if we change the number of video per page in the backend and forget to change it here?

So I guess we could use the number of video retrived in the last page for the caculations done later. 
See my next comment.

File: frontend/src/components/VideoCarrouselSlick.js [hunk 11]
```diff
@@ -35,24 +36,50 @@ class VideoCarrouselSlick extends Component {
...
+     * used to compute if we need to make new API calls
+     * @param index
+     * @returns {Promise<void>}
+     */
+    async afterChangeMethod(index) {
+        //index is gave by react slick and correspond to the index of the video on the left (start at 1)
+        const nextCarrousselCount = index > this.state.index ? this.state.carrousselCount + 1 : this.state.carrousselCount - 1;
+        //we add 5 to index to calcultate the number of videos displayed so far
+        const pageCount = (index + 5) / this.VIDEO_PER_PAGE;
+        if(pageCount === this.state.apiCallCount && pageCount <= this.state.pagesTotal){
```
Comment (xavierfav) [inline frontend/src/components/VideoCarrouselSlick.js]: We could use the number of video in the previous videos request response for calculating pageCount.
Also, we should check if we are at the end of the paginator in a different way than using `pageCount <= this.state.pagesTotal`.
I think we can, after recieving a videos response, check if we are at the end of the paginator.
The back gives num_pages. So we should just compare it to the page we requested.
If they are equal we set a flag so we know we are at the end from the front.
If I understand the variables correctly, something like (not sure if apiCallCount refers to the number of paginator page -backend- we query):
`let lastPageLoaded = ((this.apiCallCount === response.data.num_pages) ? true : false);`
Then we can put that in the state, and probably deactivate the button to slide the carousel.

**Pull request merged**